src/Uno.UI/Microsoft/UI/Xaml/Controls/TabView/TabView.cs

```diff
@@ -0,0 +1,1295 @@
+		{
+			var scrollViewer = m_scrollViewer;
+			if (scrollViewer != null)
+			{
+				var decreaseButton = m_scrollDecreaseButton;
+				var increaseButton = m_scrollIncreaseButton;
+
+				var minThreshold = 0.1;
+				var horizontalOffset = scrollViewer.HorizontalOffset;
+				var scrollableWidth = scrollViewer.ScrollableWidth;
+
+				if (Math.Abs(horizontalOffset - scrollableWidth) < minThreshold)
+				{
+					if (decreaseButton != null)
+					{
+						decreaseButton.IsEnabled = true;
+					}
+					if (increaseButton != null)
+					{
+						increaseButton.IsEnabled = false;
+					}
+				}
+				else if (Math.Abs(horizontalOffset) < minThreshold)
+				{
+					if (decreaseButton != null)
+					{
+						decreaseButton.IsEnabled = false;
+					}
+					if (increaseButton != null)
+					{
+						increaseButton.IsEnabled = true;
+					}
+				}
+				else
+				{
+					if (decreaseButton != null)
+					{
+						decreaseButton.IsEnabled = true;
+					}
+					if (increaseButton != null)
+					{
+						increaseButton.IsEnabled = true;
+					}
+				}
+			}
+		}
+
+		private void OnItemsPresenterSizeChanged(object sender, SizeChangedEventArgs args)
+		{
+			if (!m_updateTabWidthOnPointerLeave)
+			{
+				// Presenter size didn't change because of item being removed, so update manually
+				UpdateScrollViewerDecreaseAndIncreaseButtonsViewState();
+				UpdateTabWidths();
+			}
+		}
+
+		private void OnItemsChanged(object item)
+		{
+			var args = item as IVectorChangedEventArgs; //TODO:MZ:Is this appropriate cast?
+			if (args != null)
+			{
+				TabItemsChanged?.Invoke(this, args);
+
+				int numItems = TabItems.Count;
+
+				if (args.CollectionChange == CollectionChange.ItemRemoved)
+				{
+					m_updateTabWidthOnPointerLeave = true;
+					if (numItems > 0)
+					{
+						// SelectedIndex might also already be -1
+						var selectedIndex = SelectedIndex;
+						if (selectedIndex == -1 || selectedIndex == args.Index)
+						{
+							// Find the closest tab to select instead.
+							int startIndex = (int)args.Index;
+							if (startIndex >= numItems)
+							{
+								startIndex = numItems - 1;
+							}
+							int index = startIndex;
+
+							do
+							{
+								var nextItem = ContainerFromIndex(index) as ListViewItem;
+
+								if (nextItem != null && nextItem.IsEnabled && nextItem.Visibility == Visibility.Visible)
+								{
+									SelectedItem = TabItems[index];
+									break;
+								}
+
+								// try the next item
+								index++;
+								if (index >= numItems)
+								{
+									index = 0;
+								}
+							} while (index != startIndex);
+						}
+
+					}
+					// Last item removed, update sizes
+					// The index of the last element is "Size() - 1", but in TabItems, it is already removed.
+					if (TabWidthMode == TabViewWidthMode.Equal)
+					{
+						m_updateTabWidthOnPointerLeave = true;
+						if (args.Index == TabItems.Count)
+						{
+							UpdateTabWidths(true, false);
```

Comment:
Codacy found an issue: [Remove this default value assigned to parameter 'shouldUpdateWidths'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6063676)